pkg/networkmanager/wireguard.jsx

```diff
@@ -0,0 +1,353 @@
+
+    // Additional check for `wg` after install_dialog for non-packagekit and el8 environments
+    useEffect(() => {
+        async function checkWireguardPackage() {
+            try {
+                await cockpit.script("command -v wg");
+            } catch (e) {
+                setDialogError(_("wireguard-tools package is not installed"));
+            }
+        }
+
+        checkWireguardPackage();
+    }, []);
+
+    useEffect(() => {
+        if (!connection) generatePrivateKey();
+    }, [connection]);
+
+    useEffect(() => {
+        async function getPrivateKey() {
+            const objpath = connection[" priv"].path;
+            const [result] = await model.client.call(objpath, "org.freedesktop.NetworkManager.Settings.Connection", "GetSecrets", ["wireguard"]);
+            setGeneratedPrivateKey(result.wireguard["private-key"].v);
+        }
+
+        if (connection?.[" priv"].path) {
+            getPrivateKey();
+        }
+    }, [model, connection]);
+
+    useEffect(() => {
+        const privateKey = isPrivKeyGenerated ? generatedPrivateKey : pastedPrivateKey;
+        if (privateKey === "") {
+            setPublicKey("");
+            return;
+        }
+
+        async function getPublicKey() {
+            try {
+                const key = await cockpit.spawn(["wg", "pubkey"], { err: 'message' }).input(privateKey.trim());
+                setPublicKey(key.trim());
+            } catch (e) {
+                console.error(e.message);
+                setPublicKey('');
+            }
+        }
+
+        getPublicKey();
+    }, [isPrivKeyGenerated, generatedPrivateKey, pastedPrivateKey]);
+
+    async function generatePrivateKey() {
+        try {
+            const key = await cockpit.spawn(["wg", "genkey"]);
+            setGeneratedPrivateKey(key.trim());
+        } catch (e) {
+            setDialogError(e.message);
+        }
+    }
+
+    function onSubmit() {
+        const private_key = isPrivKeyGenerated ? generatedPrivateKey : pastedPrivateKey;
+
+        // Validate Addresses before submit
+        // Also validate listenPort as PF TextInput[type=number] accepts normal text as well on firefox
+        // See - https://github.com/patternfly/patternfly-react/issues/9391
+        let addr, peersArr;
+        const listen_port = Number(listenPort);
+        try {
+            addr = stringToAddresses(addresses);
+
+            if (isNaN(listen_port)) {
+                throw new Error(_("Listen port must be a number"));
+            }
+
+            peersArr = peers.map((peer, index) => {
+                if (peer.endpoint !== '') {
+                    const parts = peer.endpoint.split(":");
+                    if (parts.length !== 2) {
+                        throw cockpit.format(_("Peer #$0 has invalid endpoint. It must be specified as host:port, e.g. 1.2.3.4:51820 or example.com:51820"), index + 1);
```

Comment:
These 2 added lines are not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19024-20230905-070545-69e0670b-fedora-38-devel//Coverage/lcov/github-pr.diff.gcov.html)